selene/schema/results/fields.py

```diff
@@ -97,35 +126,49 @@ class Meta:
 
     description = graphene.String(description='Description of the result')
 
-    detection_result = graphene.Field(
-        DetectionResult, description='Detection result'
+    origin_result = graphene.Field(
+        OriginResult,
+        description='Referenced result that provided information for creating '
+        'this result',
     )
 
-    report_id = graphene.UUID(description="ID of the corresponding report")
+    result_type = graphene.Field(
+        ResultType,
+        name='type',
+        description='Type of result. Currently it can be a NVT or CVE based '
+        'result',
+    )
+
+    report = graphene.Field(
+        ResultReport, description='Report the result belongs to'
+    )
     task = graphene.Field(ResultTask, description='Task the result belongs to')
     host = graphene.Field(ResultHost, description='Host the result belongs to')
-    port = graphene.String(description='The port on the host')
```

Comment:
As I wrote in the commit message and the PR header the field doesn't contain always a port. It may contain other things. One thing that clearly comes to my mind is for example `general/tcp`. For me the name port feels wrong. 

If I remember correctly the naming change from port to location has been requested several years ago but just wasn't completely implemented.

Reply:
Still though `general/tcp` is representing port(s) here. :)